src/redisraft.c

```diff
@@ -591,76 +591,79 @@ static bool handleInterceptedCommands(RedisRaftCtx *rr,
-        return RR_ERROR;
-    }
-
-    if (!raft_is_leader(rr->raft)) {
-        Node *leader = getLeaderNodeOrReply(rr, ctx);
-        if (!leader) {
-            return RR_ERROR;
-        }
-
-        return handleNonLeaderCommand(rr, ctx, leader, cmds);
-    }
-
-    /* Handle the special case of read-only commands here: if quorum reads
-     * are enabled schedule the request to be processed when we have a guarantee
-     * we're still a leader. Otherwise, just process the reads.
-     *
-     * Normally we can expect a single command in the request, unless it is a
-     * MULTI/EXEC transaction in which case all queued commands are handled at
-     * once.
-     */
-    unsigned int cmd_flags = CommandSpecGetAggregateFlags(cmds, CMD_SPEC_WRITE);
-    if (cmd_flags & CMD_SPEC_UNSUPPORTED) {
-        RedisModule_ReplyWithError(ctx, "ERR not supported by RedisRaft");
-        return RR_ERROR;
-    } else if (cmd_flags & CMD_SPEC_READONLY && !(cmd_flags & CMD_SPEC_WRITE)) {
-        if (rr->config->quorum_reads) {
-            RaftReq *req = RaftReqInit(ctx, RR_REDISCOMMAND);
-            RaftRedisCommandArrayMove(&req->r.redis.cmds, cmds);
-            raft_queue_read_request(rr->raft, handleReadOnlyCommand, req);
-        } else {
-            /* Wait until the new leader applies an entry from the current term.
-             * Otherwise, we might process a request before replaying logs.
-             * The state machine will be in an older state. Reading from it
-             * might look like going backward in time. */
-            raft_term_t term = raft_get_current_term(rr->raft);
-            if (term != rr->snapshot_info.last_applied_term) {
-                replyClusterDown(ctx);
-                return RR_ERROR;
-            }
-
+/* Handle the special case of read-only commands here: if quorum reads
+ * are enabled schedule the request to be processed when we have a guarantee
+ * we're still a leader. Otherwise, just process the reads.
+ *
+ * Normally we can expect a single command in the request, unless it is a
+ * MULTI/EXEC transaction in which case all queued commands are handled at
+ * once.
+ */
+static void handleRedisCommandAppendReadOnly(RedisRaftCtx *rr, RedisModuleCtx *ctx,
+                                             RaftRedisCommandArray *cmds) {/* read only command, quorum or not */
+    if (rr->config->quorum_reads) {
+        RaftReq *req = RaftReqInit(ctx, RR_REDISCOMMAND);
+        RaftRedisCommandArrayMove(&req->r.redis.cmds, cmds);
+        raft_queue_read_request(rr->raft, handleReadOnlyCommand, req);
+    } else {
+        /* Wait until the new leader applies an entry from the current term.
+         * Otherwise, we might process a request before replaying logs.
+         * The state machine will be in an older state. Reading from it
+         * might look like going backward in time. */
+        raft_term_t term = raft_get_current_term(rr->raft);
+        if (term == rr->snapshot_info.last_applied_term) {
```

Comment:
I think this makes code less readable. Common pattern is having an if check for error case and returning inside it. So, you keep success path in the main flow. Previous version of this code is easier to understand.

Reply:
ok, changed back, was trying to avoid extra returns